docs/reference/modules/http.asciidoc

```diff
@@ -108,3 +108,33 @@ client HTTP responses, defaults to unbounded.
 
 It also uses the common
 <<modules-network,network settings>>.
+
+[float]
+=== Rest Request Tracer
+
+The http module has a dedicated tracer logger which, when activated, logs incoming requests. The log can be dynamically activated
+by setting the level of the `org.elasticsearch.http.AbstractHttpServerTransport.tracer` logger to `TRACE`:
+
+[source,console]
+--------------------------------------------------
+PUT _cluster/settings
+{
+   "transient" : {
+      "logger.org.elasticsearch.http.AbstractHttpServerTransport.tracer" : "TRACE"
```

Comment:
I wonder if we should rename `AbstractHttpServerTransport` to `HttpServerTransport`, so that this looks less ugly. @tbrooks8 thoughts?

Reply:
I'm all for the renaming. We could technically even use a logger name not tied to a specific class name couldn't we?
I'm wondering if `logger.org.elasticsearch.http.tracer` isn't even better?

Reply:
I am very much in favor of dedicated name. There is already a little bit of tension between `TransportService.tracer` and `TransportLogger`. So I think we are going to need to unify those at some point. I think that we should pick a natural name for the http logger which might be a guide for a unified transport tracer/logger.

I feel a little weird about `logger.org.elasticsearch.http.tracer` as it seems kind of odd when all other logs will include a class. `org.elasticsearch.http.Tracer`? `org.elasticsearch.http.HttpTracer`? I'm not super opposed to `logger.org.elasticsearch.http.tracer`. It just feels weird (even if it kind of looks like what we currently do for the TransportService).

Reply:
+1 to factor out the tracing functionality into its own class (e.g. org.elasticsearch.http.HttpTracer), which will serve as name for the logger to be configured.

Reply:
@ywelsch @tbrooks8 alright, extracted the whole trace logging logic into its own class in https://github.com/elastic/elasticsearch/pull/51684/commits/bf325bae194a5f8641458a989f9cc599d2855856 :)